server/src/middleware/auth/helpers/register/add-unique-code-to-user.js

```diff
@@ -0,0 +1,18 @@
+const generateUniqueCode = require('./generate-unique-code');
+const uniqueCodeExistsOnDB = require('../../../helpers/unique-code-exists');
+
+async function addUniqueCodeToUser(user) {
+  let code = generateUniqueCode();
+
+  if (await uniqueCodeExistsOnDB(code)) {
+    // Recursively call this function until generated code is unique
+    return addUniqueCodeToUser(user);
```

Comment:
The probability of generating collisions is currently pretty low and will slowly increase with each user. Because of this, it will only check the DB once (most of the time). I could see this becoming a problem with millions of users though (which I'm not designing for right now).